### ISSUE TITLE
Add action typing

### ISSUE THREAD

**Issue by krzema12** [repo-sync/pull-request#131]
I'm looking to get your action first-class supported by https://github.com/krzema12/github-workflows-kt, a Kotlin DSL to write GitHub Action workflows.

They came up with a way to reduce operational load when keeping library's action wrappers in sync with actions' inputs and outputs. The solution includes onboarding https://github.com/krzema12/github-actions-typing. It's as easy as adding an extra YAML file to your repository root, and adding a simple GitHub workflow that validates this new file. Thanks to this, the code generator in the Kotlin DSL can fetch typing info provided by you instead of them, which has a number of benefits. It has no negative effects on current action consumers, they continue to use the action via regular GitHub API, as if the file wasn't there.

In this feature request, I would like to ask you if you're open to introducing such typings in your action. You wouldn't be first - there're already other actions using it, like e.g.: https://github.com/Vampire/setup-wsl or https://github.com/ReactiveCircus/android-emulator-runner (full list [here](https://github.com/krzema12/github-actions-typing/network/dependents)).

If your answer is "yes", feel free to either add it yourself, or let me know - me or some of the library contributors would be happy to post a PR. They're also open to any kind of questions and feedback.

**Comment by m4heshd** [comment]
Hi, pinging @wei since they're the owner of this repo.

**Comment by heiskr** [comment]
This is a wonderful idea. I don't think we would support this unless it became a built-in Actions feature, however.

**Comment by krzema12** [comment]
@heiskr I don't think it will happen within the next year or two - GitHub is not really interested in implementing it. That's why  I decided to create and maintain the library.

**Comment by heiskr** [comment]
Thank you for opening this issue. Per the updated README, we are in process of archiving this repository, and encourage using [`gh pr create`](https://cli.github.com/manual/gh_pr_create) directly in your workflows instead.
